s32/README

```diff
@@ -38,7 +38,6 @@ License Link:
    https://spdx.org/licenses/BSD-3-Clause.html
 
 Patch List:
```

Comment:
I think this is a good time to update this list of patches. Pls add a sub-section for S32Z/E and add a mention to each of the the patches applied to the original driver sources. The ones that were applied only to S32K3 move it to another sub-section.

Reply:
I updated to reduce num of commits and add 2 sub section list of patches for s32ze and s32k3